### DISCUSSION TITLE
xdg-open Process Blocking Terminal on i3

### DISCUSSION THREAD

**Discussion by newptcai** [i3/i3#6217] [Q&A] [answered]
I've always used `xdg-open` to open files, like PDFs, directly from the terminal. On Ubuntu and Regolith, when I run this command, the process to display the file content automatically detaches from the terminal, allowing me to continue using the terminal without any issues. However, on i3, the process seems to block the terminal until I close the file. It's not a major issue, but I'm curious—what might be causing this difference in behavior? 

Thanks for any insights!

**Comment by orestisfl** [comment]
Works as expected on my system, which distro are you on?

**Reply by newptcai** [reply]
I install i3 through Ubuntu 24.04. Usually I use Regolith, but it is not available for 24.04 yet. So I switched to use i3 directly.

**Comment by newptcai** [comment, marked as answer]
It turns out I need to change environment variable by adding this to `~/.profile`
```
export XDG_CURRENT_DESKTOP=GNOME
```

**Comment by newptcai** [comment]
Once I fixed this, it becomes very slow in opening things like a file picker in Firefox. :cry:
